src/main/java/net/neoforged/neoforge/client/model/AbstractUnbakedModel.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.model;
+
+import net.minecraft.client.renderer.block.model.ItemTransforms;
+import net.minecraft.client.renderer.block.model.TextureSlots;
+import net.minecraft.client.resources.model.UnbakedModel;
+import net.minecraft.util.context.ContextMap;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Base unbaked model for custom models which support the standard top-level model parameters
+ * added by vanilla and NeoForge except elements but create the quads from something other
+ * than the vanilla elements spec.
+ */
+public abstract class AbstractUnbakedModel implements ExtendedUnbakedModel {
```

Comment:
Should there be a bake method that passes the `StandardModelParameters`? Not sure because the nullability of many fields is different. We just have to clearly indicate that the `parameters` field should not be used from `bake`. Instead, the information should be pulled from the parameters to `bake`.

Reply:
I don't think a special `bake()` method makes sense. I'll add a comment to the `parameters` field though noting that it should not be used from `bake()`.